src/tokens.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import { _ } from 'lodash'
+import type {
+  DbApi,
+  Token,
+  WithTokens,
+  TokenWithDbId,
+} from 'icarus-backend'; // eslint-disable-line
+import { arraySum } from './helpers'
+
+const formatToken = (token: TokenWithDbId | Token): Token => ({
+  policyId: token.policyId,
+  assetName: token.assetName,
+  quantity: token.quantity,
+})
+
+const aggregateTokensForOnePolicy = (policyGroup: any, policyId: string) => _(policyGroup)
+  .groupBy(({ assetName }) => assetName)
+  .map((assetGroup, assetName) => formatToken({
+    policyId,
+    assetName,
+    quantity: `${arraySum(assetGroup.map(asset => asset.quantity))}`,
+  }))
+  .value()
+
+export const aggregateTokens = (
+  tokens: Array<Array<Token> | null>,
+): Array<Token> | null => {
+  const aggregatedTokens = _(tokens)
+    .filter(token => token !== null)
+    .flatten()
+    .groupBy(({ policyId }) => policyId)
+    .map(aggregateTokensForOnePolicy)
+    .flatten()
+    .value()
+  return aggregatedTokens.length ? aggregatedTokens : null
```

Comment:
any specific reason to return null? Wouldn't an empty array be expressive enough?

Reply:
Makes sense. It would also help clear all the `<Token> | null` expressions.